File: dev/test_data/main/typescript/expected/aas_core_meta.v3/input/snippets/Verification/qualifier_types_are_unique.ts
/**
 * Check that there are no duplication {@link types.Qualifier.type}'s
 * in the `qualifiers`.
 *
 * @param qualifiers - to be verified
 * @returns `true` if the check passes
 */
export function qualifierTypesAreUnique(
  qualifiers: Iterable<AasTypes.Qualifier>
): boolean {
  const typeSet = new Set<string>();

  for (const qualifier of qualifiers) {
    if (typeSet.has(qualifier.type)) {
      return false;
    }

    typeSet.add(qualifier.type);
  }

  return true;
}

File: dev/test_data/main/python/expected/aas_core_meta.v3/input/snippets/Verification/qualifier_types_are_unique.py
def qualifier_types_are_unique(qualifiers: Iterable[aas_types.Qualifier]) -> bool:
    """
    Check that there are no duplicate
    :py:attr:`.types.Qualifier.type`'s
    in the :paramref:`qualifiers`.
    """
    type_set = set()  # type: Set[str]
    for qualifier in qualifiers:
        if qualifier.type in type_set:
            return False

        type_set.add(qualifier.type)

    return True
